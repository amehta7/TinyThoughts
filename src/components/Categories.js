import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const Categories = ({ categories }) => {
  return (
    <ul className='sidebar'>
      {categories &&
        categories.map(({ id, title }) => {
          return (
            <li key={id}>
              <Link to={`/category/${id}`} key={id}>
                {title}
              </Link>
            </li>
          )
        })}

      <Link to='/'>
        <li>All Posts</li>
      </Link>
    </ul>
  )
}

Categories.propTypes = {
  title: PropTypes.string,
}

export default Categories
