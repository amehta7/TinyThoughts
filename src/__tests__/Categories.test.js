import React from "react";
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import Categories from "../components/Categories";

describe("Categories component", () => {
  it("Categories component renders a list of categories", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Categories
          categories={[
            { id: 1, title: "JavaScript" },
            { id: 2, title: "Python" }
          ]}
        />
      </MemoryRouter>
    );

    expect(wrapper.find(Categories)).toMatchSnapshot();
    expect(
      wrapper
        .find("a")
        .at(1)
        .prop("href")
    ).toEqual("/category/2");
  });
});
