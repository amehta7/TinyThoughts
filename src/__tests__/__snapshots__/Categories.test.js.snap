// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component Categories component renders a list of categories 1`] = `
<Categories
  categories={
    Array [
      Object {
        "id": 1,
        "title": "JavaScript",
      },
      Object {
        "id": 2,
        "title": "Python",
      },
    ]
  }
>
  <ul
    className="sidebar"
  >
    <li
      key="1"
    >
      <Link
        key="1"
        to="/category/1"
      >
        <LinkAnchor
          href="/category/1"
          navigate={[Function]}
        >
          <a
            href="/category/1"
            onClick={[Function]}
          >
            JavaScript
          </a>
        </LinkAnchor>
      </Link>
    </li>
    <li
      key="2"
    >
      <Link
        key="2"
        to="/category/2"
      >
        <LinkAnchor
          href="/category/2"
          navigate={[Function]}
        >
          <a
            href="/category/2"
            onClick={[Function]}
          >
            Python
          </a>
        </LinkAnchor>
      </Link>
    </li>
    <Link
      to="/"
    >
      <LinkAnchor
        href="/"
        navigate={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <li>
            All Posts
          </li>
        </a>
      </LinkAnchor>
    </Link>
  </ul>
</Categories>
`;

exports[`Categories component renders a list of categories`] = `
<Categories
  categories={
    Array [
      Object {
        "id": 1,
        "title": "JavaScript",
      },
      Object {
        "id": 2,
        "title": "Python",
      },
    ]
  }
>
  <ul
    className="sidebar"
  >
    <Link
      key="1"
      to="/category/1"
    >
      <LinkAnchor
        href="/category/1"
        navigate={[Function]}
      >
        <a
          href="/category/1"
          onClick={[Function]}
        >
          <li>
            JavaScript
          </li>
        </a>
      </LinkAnchor>
    </Link>
    <Link
      key="2"
      to="/category/2"
    >
      <LinkAnchor
        href="/category/2"
        navigate={[Function]}
      >
        <a
          href="/category/2"
          onClick={[Function]}
        >
          <li>
            Python
          </li>
        </a>
      </LinkAnchor>
    </Link>
    <Link
      to="/"
    >
      <LinkAnchor
        href="/"
        navigate={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <li>
            All Posts
          </li>
        </a>
      </LinkAnchor>
    </Link>
  </ul>
</Categories>
`;
